### DISCUSSION TITLE
Import data from dump file when creating new database

### DISCUSSION THREAD

**Discussion by wagich** [ravendb/ravendb#12735] [Ideas]
Quite often I'll need to spin up a copy of a production database locally for general development or debugging. Our DBs very seldom include sensitive data or PII (and if they do, the collections can be excluded when exporting) so that's a very convenient way to get up-to-date data.

Right now this involves exporting the data to a dump file, locally (re)creating the database, then importing the file.
It would be nice if the database dump file could be specified (and uploaded) when creating the new database.

A similar functionality exists for restoring from backups, but it only works with files on the server or in the cloud and only with a "proper" backup.

**Comment by garayx** [comment]
Hi 

Thanks for you suggestion, its only supported to create an empty database, or create it from a backup file.

You may want to create a manual backup (https://ravendb.net/docs/article-page/5.2/csharp/studio/database/tasks/backup-task#creating-manual-backups) and use it to create a new database.

**Comment by ayende** [comment]
You can also setup an import from another live instance, so the process is create a new db, and then import from production.

There is also the ability to run ETL from production to test/dev as an ongoing project.

**Comment by wagich** [comment]
@garayx  Yeah, I figured. Creating a backup is a bit more involved though – the nice thing about a dump is that it is directly downloaded to my machine and I don't need to create a backup somewhere and then go fetch it manually.
But now that I think about it, creating a cloud location to backup to and then directly restore from might be a good alternative… I'm gonna try that, thanks!

@ayende I was thinking about these options, but wouldn't live-importing from another instance require that the prod-cluster trusts dev?
I didn't really consider ETL because it needs per-db setup (and we have a lot of dbs) and would require dev to be reachable from prod in some way.

**Reply by ayende** [reply]
You will need a path from dev to prod or vice versa. We have a bunch of features that may help here, but in all cases, one or the other need to be able to connect.

If you have totally disconnected environments, export / import is the way to go, yes.
